nix/mayastor-overlay.nix

```diff
@@ -3,11 +3,13 @@ self: super: {
   nvmet-cli = super.callPackage ./pkgs/nvmet-cli { };
   libspdk = (super.callPackage ./pkgs/libspdk { }).release;
   libspdk-dev = (super.callPackage ./pkgs/libspdk { }).debug;
-  mayastor = (super.callPackage ./pkgs/mayastor { }).release;
-  mayastor-dev = (super.callPackage ./pkgs/mayastor { }).debug;
-  mayastor-adhoc = (super.callPackage ./pkgs/mayastor { }).adhoc;
+  mayastor-release = (super.callPackage ./pkgs/mayastor { }).release;
```

Comment:
I think keeping `mayastor` as the canonical name is worthwhile. 

Based on my current understanding of Nix, It is not idiomatic to have `-dev` postfixes in Nix. It seems like this is configured via `--argstr` usually? I feel like this is for future work, though.

Reply:
I wanted to keep `mayastor` as the name here but because the name in `images/default.nix` was changed from `mayastor-image` to `mayastor` when you try to build it using `nix-build -A images.mayastor` you get an error message indicating `infinite recursion encountered`

Reply:
I figured out how to keep what we already had, so it is back to `mayastor` as the canonical name.